### ISSUE TITLE
Project settings are not applied

### ISSUE THREAD

**Issue by jakenvac** [migueldeicaza/SwiftGodotKit#10]
Firstly, thanks for this project. I'm having a lot of fun with it.

As the issue title states, project settings are not applied. Specifically the `loadProjectSettings` callback in `runGodot` is never called.

I'm pretty sure this is because in `SwiftGodotKit.swift` the call to `projectSettingsBind` has been commented out. I'll submit a PR to restore this, but I thought I'd raise an issue for discussion in case it was commented out for a reason.

Here's a minimal example to show the setting is never applied and the print statements are never printed:
```swift
import Foundation
import SwiftGodotKit
import SwiftGodot

let SETTING_KEY = "display/window/size/borderless"

func loadScene (scene: SceneTree) {
    let gui = Gui()
    scene.root?.addChild(node: gui)
}

class Gui: Container {
    public required init (nativeHandle: UnsafeRawPointer) {
        super.init (nativeHandle: nativeHandle)
    }
    
    public required init () {
        super.init ()
    }

    public override func _ready() {
        let setting = ProjectSettings.shared.getSetting(name: SETTING_KEY, defaultValue: Variant("junk"))
        print("Setting: ", Bool(setting)!)
    }
}

func registerTypes (level: GDExtension.InitializationLevel) {
    switch level {
    case .scene:
        register (type: Gui.self)
    default:
        break
    }
}

@main
class Startup {
    static func main () {
        runGodot(args: [], initHook: registerTypes, loadScene: loadScene, loadProjectSettings: { settings in
            print("Hello world")
            GD.print("Hello godot")
            settings.setSetting(name: SETTING_KEY, value: Variant(true))
        })
    }
}
```

**Comment by migueldeicaza** [comment]
Mhm, I think that what happened here is that the libgodot patch used to have a way of calling this project setup method, and at some point it got dropped.   Because the issue is that the libgodot_bind call does not exist.

The unfortunate part is that I no longer know where that code went, and the discussion that covers this merely states that the code will be submitted at some point in the future:

https://github.com/godotengine/godot/pull/72883

Since ProjectSettings is a singleton, I wonder if you could try to access that data from the registerTypes callback and if it has enough information for you to proceed at that point.

**Comment by migueldeicaza** [comment]
Yes, let us keep this issue open until we figure this out - I am sorry I did not catch this earlier.

**Comment by BadChoice** [comment]
Just to expand it, adding a `project.godot` file in the root, works, the settings are used by the engine 
<img width="268" alt="image" src="https://github.com/migueldeicaza/SwiftGodotKit/assets/93900/224b2cc0-ff56-4dbc-a3af-13c736119321">

**Comment by migueldeicaza** [comment]
Thanks for this tip @BadChoice - @jakenvac could you confirm if this fixes the issue for you?

In that case, I should document that

**Comment by migueldeicaza** [comment]
swiftPM or Xcode build?  What I was fighting last night was trying to get the resources copied in the right place, so would love a simple reproduction test case so I can dig into that specific setup
